bundles/mapping/drawtools/plugin/DrawPlugin.ol.js

```diff
@@ -998,56 +808,70 @@ Oskari.clazz.define(
          * @param {String} shape
          * @param {Object} options
          */
-        addModifyInteraction: function (layerId, shape) {
-            const me = this;
-            const layer = me.getLayer(layerId);
-            if (layer) {
-                me._modify[me._id] = new olInteractionModify({
-                    features: layer.getSource().getFeaturesCollection(),
-                    style: me._styles.modify,
-                    insertVertexCondition: isModifyLimited(shape) ? olEventsCondition.never : olEventsCondition.always,
-                    deleteCondition: function (event) {
-                        return olEventsCondition.shiftKeyOnly(event) && olEventsCondition.singleClick(event);
-                    }
-                });
-            }
-            me.modifyStartEvent(shape);
-            me.getMap().addInteraction(me._modify[me._id]);
+        addModifyInteraction: function () {
+            const isLimited = isModifyLimited(this.getShape());
+            this._modify = new olInteractionModify({
+                source: this.getDrawSource(),
+                style: this.getPointerStyleFunction(),
+                insertVertexCondition: isLimited ? olEventsCondition.never : olEventsCondition.always,
+                deleteCondition: isLimited ? olEventsCondition.never : event => olEventsCondition.shiftKeyOnly(event) && olEventsCondition.singleClick(event)
```

Comment:
I haven't noticed this deleteCondition config before so I tested it out with current version. Drawing a line with start and having shift pressed while clicking the next makes the measurement tooltip stay on screen while removing the line. Should we have this olEventsCondition.never in all conditions instead?

Reply:
![drawwithshift](https://user-images.githubusercontent.com/2210335/206756563-2fb0d352-0f16-4ac8-93e9-d17b95bf55e6.gif)

Here on the first line I shift click the last point. On second line I finish it with double click. On third using shift click again. For some reason shift with dragging goes to the "freeform" draw, but just clicking with shift pressed does this.

Reply:
We should test if having `deleteCondition` always on `olEventsCondition.never` fixes this and maybe leave it on never.